### DISCUSSION TITLE
Use Crossplane with Postman

### DISCUSSION THREAD

**Discussion by saichaitanya11** [crossplane/crossplane#3875] [Q&A]
Hello,

Is there a way to use Crossplane APIs using Postman calls or through Python requests? Can you please share the documentation and process on how create our own resources, update and delete them as and when required using Postman requests, so that these requests can further be integrated into a UI. Please let me know how to authenticate the requests too. 

Thank you

**Comment by bobh66** [comment]
At the REST API level which Postman supports Crossplane is using Kubernetes APIs, so you can do something like this:

https://iximiuz.com/en/posts/kubernetes-api-call-simple-http-client/

https://www.postman.com/api-evangelist/workspace/kubernetes/overview

**Reply by saichaitanya11** [reply]
@bobh66 Thank you so much for the quick response. I will check on the resources and get back to you if I have any doubts.

Thank you

**Reply by saichaitanya11** [reply]
Hi @bobh66 ,

I went through the resources shared by you. Thank you, they have been very helpful. But I couldn't find a way to create a resource in AWS using Crossplane via these APIs. For example, if I want to create an S3 Bucket, I have a YAML script and I use _kubectl apply -f "filename.yaml"_ to create the resource. How do I replicate the same using Kubernetes/Crossplane APIs in Postman?

I used the following API
_$KUBE_SERVER/apis/apps/v1/namespaces/default/deployments_ as a _POST_ request and passed the same script which I used to create a bucket using kubectl in the JSON format in the body. Got the following error

_"status": "Failure",
    "message": "Bucket in version \"v1beta1\" cannot be handled as a Deployment: no kind \"Bucket\" is registered for version \"s3.aws.upbound.io/v1beta1\" in scheme \"pkg/api/legacyscheme/scheme.go:30\"",
    "reason": "BadRequest",
    "code": 400_

Can you please guide in the direction which helps me create or delete a Crossplane resource using Postman. 
Thank you

**Comment by bobh66** [comment]
You can use the "-v=9" option to your `kubectl apply -f` command to see the specific REST API calls that kubectl makes to create the object.  That should help with the content you need to use in Postman.

**Reply by saichaitanya11** [reply]
Hi @bobh66 ,

These solutions have been extremely helpful. Thank you so much for these resources. I have one question. To authorize these APIs, we would need a bearer token which I am getting by running the `kubectl create token default` command in the shell and using it as an authorization header for my requests. The API corresponding to that command is `$KUBE_SERVER/api/v1/namespaces/default/serviceaccounts/default/token`. When I run this is in Postman, this is the response I get. 
`{
    "kind": "Status",
    "apiVersion": "v1",
    "metadata": {},
    "status": "Failure",
    "message": "serviceaccounts \"default\" is forbidden: User \"system:anonymous\" cannot create resource \"serviceaccounts/token\" in API group \"\" in the namespace \"default\"",
    "reason": "Forbidden",
    "details": {
        "name": "default",
        "kind": "serviceaccounts"
    },
    "code": 403
}`

How can I authenticate the token API in Postman so that I don't have to go the shell each time to get a new token?

Thank you

**Reply by bobh66** [reply]
That will depend on the type of kubernetes cluster you are using and the source of the tokens.  For example if you are using AWS EKS you would need to implement the REST API call(s) for getting a valid EKS token to use in your kubernetes API calls.  Sorry I can't be of much help with this area.
